Replace ENV VAR 'MASTER_TAG' by 'TAG' in build sh scripts

chrismayer opened the pull request:
This replaces the hard coded var 'MASTER_TAG' by the env var `TAG` in the build bash scripts. So the tag for the Docker images can be modified from outside and the `TAG` env var used in the `docker-stack.yml` can be reused.
The default is `'master' `as it was the hard coded value before, so no functional change at all (if nothing else is configured).

This allows to build and fire up the Docker stack with a consitent TAG for all Docker images by executing:

`TAG=kalle ./build-and-start.sh`

Very helpful for local dev setup or publishing all images with a dedicated tag.

JakobMiksch commented:
@chrismayer Does calling `TAG=kalle ./build-and-start.sh` really overwrite the `TAG` set in the script. For me it looks like the `TAG` will be set in the script again.

JakobMiksch commented:
@chrismayer nevermind. I just saw, that you are actually calling `TAG` in the assignment again `TAG="${TAG:-master}"`

JakobMiksch reviewed (changes requested):
just checked it. For me it looks like to use `export` for the variable assignment:

```
export TAG=PETER; bash build-and-start.sh 
```

otherwise the assignment does not have effect

JakobMiksch commented:
A quick note in the README would be great as well

chrismayer commented:
Added the missing forwarding of the TAG var to the internally called script `build-images.sh`. So now executing `TAG=kalle ./build-and-start.sh` should work. Of course you can do `export TAG=peter; bash build-and-start.sh `, which sets the ENV VAR temp. for the whole system, which be a bit too much (depends on the case).

I also added some information to the README.

jdk24 reviewed (approved):
Tested successfully via changes in local image and using custom tag for `TAG=testing ./build-and-start.sh`. 
Changes were applied and services started successfully with custom tag. 
Also correctly defaults to master tag when no tag parameter is given. 
Thanky you and please merge at will!

JakobMiksch commented:
@chrismayer @jdk24 
It basically runs, but I get two errors in the build. I am not sure if this is related:
- on for Raster-Download which can be fixed by adding `RUN chmod 775 /copy_pg_bins.sh` on line 3 of the Dockerfile
- in um_ol_demo a large Python error

chrismayer commented:
@JakobMiksch Yes, seems to be a problem with your setup (or something similar). At least not related to this PR. 

I'll merge now. Thanks for your reviews and testing!
